mllib/src/test/scala/org/apache/spark/ml/classification/RandomForestClassifierSuite.scala

```diff
@@ -178,6 +179,34 @@ class RandomForestClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     assert(importances.toArray.forall(_ >= 0.0))
   }
 
+  test("should support all NumericType labels") {
+    val dfs = MLTestingUtils.genClassifDFWithNumericLabelCol(sqlContext, "label", "features")
+
+    val rf = new RandomForestClassifier().setFeaturesCol("features")
+
+    val expected = rf.setLabelCol("label")
+      .fit(TreeTests.setMetadata(dfs(DoubleType), 2, "label"))
+    dfs.keys.filter(_ != DoubleType).foreach { t =>
+      TreeTests.checkEqual(expected,
+        rf.setLabelCol("label").fit(TreeTests.setMetadata(dfs(t), 2, "label")))
+    }
+  }
+
+  test("shouldn't support non NumericType labels") {
```

Comment:
I wanted to be able to do something like:

```scala
def checkNumericTypes[T <: Regressor, M <: Model](
    regressor: T, sqlContext: SQLContext)(asserts: Seq[(M, M) => Unit]): Unit = {
  val dfs = MLTestingUtils.genRegressionDFWithNumericLabelCol(sqlContext, "label", "features")
    
  val expected = regressor.fit(dfs(DoubleType))
  dfs.keys.filter(_ != DoubleType).foreach { t =>
    val actual = regressor.fit(dfs(t))
    asserts.foreach(f => f(expected, actual))
  }
}
```

But unfortunately that doesn't work since I'm not able to capture which model is being created by the regressor's `fit` method.